src/Providers/ServiceProvider.php

```diff
@@ -22,9 +24,19 @@ public function boot()
             __DIR__.'/../../config/kinesis.php' => config_path('kinesis.php'),
         ]);
 
-        if (config('kinesis.stream')) {
-            $monolog = Log::getLogger();
+        if ($this->app['log'] instanceof LogManager) {
+            $this->app['log']->extend('kinesis', function (Container $app, array $config) {
+                $handler = new KinesisHandler($config['stream'], $config['level'] ?? LogLevel::INFO);
 
+                return new Logger('kinesis', [$handler]);
+            });
+        }
+
+        app()->bind(KinesisFormatter::class, function () {
+            return new KinesisFormatter(config('app.name'), $this->app->environment());
+        });
+
+        app()->bind(KinesisClient::class, function () {
             $config = [
                 'region' => config('kinesis.aws.region'),
                 'version' => 'latest'
```

Comment:
Can we update the `config/kinesis.php` to only have the AWS crednetials as we no longer specify the stream or log level in there but directly within the `logging.php` of the consuming application.